src/internal/storage/server_test.go

```diff
@@ -255,6 +255,18 @@ func createFileset(ctx context.Context, c storage.FilesetClient, num, size int)
 				return "", nil, err
 			}
 		}
+		if size == 0 {
```

Comment:
A cleaner way to do this imo is to always send an initial message with zero bytes before the chunking above.